src/main/java/com/github/princesslana/eriscasper/BotContext.java

```diff
@@ -37,7 +37,9 @@ public BotContext(Observable<Event> events, Routes routes, RepositoryManager rep
 
   public <D, E extends Event & Wrapper<D>> Completable on(
       Class<E> evt, Function<D, Completable> f) {
-    return events.ofType(evt).map(Wrapper::unwrap).flatMapCompletable(f::apply);
+    // It would be nice to use a method reference here,
+    // but doing so causes an exception at runtime
+    return events.ofType(evt).map(e -> e.unwrap()).flatMapCompletable(f::apply);
```

Comment:
Use `E::unwrap` not `Wrapper::unwrap`. Using E would source the generic not the wrapper. If it still produces a runtime it would be really weird since it must be the generic when pushed into from the class reference.

Reply:
I tried `E::unwrap` it also suffered the same issue. I have theories about why `::unwrap` approaches aren't working, but I haven't been able to recreate with a minimal example.

Reply:
Added a test for it that recreates the exception with `Wrapper::unwrap` or `E::unwrap`

Reply:
interesting